tools/roxvet/analyzers/dontcompareproto/analyzer.go

```diff
@@ -0,0 +1,124 @@
+package donotcompareproto
+
+import (
+	"fmt"
+	"go/ast"
+	"go/types"
+	"strings"
+
+	"github.com/stackrox/rox/pkg/set"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+	"golang.org/x/tools/go/types/typeutil"
+)
+
+const doc = `Inspect calls to Equal for proto arguments that should be compared with protocompat.Equal instead`
+
+// Analyzer is the go vet entrypoint
+var Analyzer = &analysis.Analyzer{
+	Name:     "donotcompareproto",
+	Doc:      doc,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+	Run:      run,
+}
+
+var (
+	protoPkgs = []string{
+		"*github.com/stackrox/scanner/generated",
+		"*github.com/stackrox/rox/generated/",
+	}
+
+	replacements = map[string]string{
+		"":              "Equal",
+		"[]":            "SlicesEqual",
+		"map[string][]": "MapSliceEqual",
+		"map[string]":   "MapEqual",
+	}
+
+	allowedCallerPackages = []string{
+		"github.com/stackrox/rox",
+	}
+
+	bannedEqualFunctions = set.NewFrozenStringSet(
+	//"github.com/google/go-cmp/cmp.Equal",
+	//"reflect.DeepEqual",
+	)
+
+	bannedAssertFunctions = set.NewFrozenStringSet(
+	//"(*github.com/stretchr/testify/assert.Assertions).Contains",
+	//"(*github.com/stretchr/testify/assert.Assertions).Equal",
+	//"(*github.com/stretchr/testify/assert.Assertions).NotEqual",
+	//"(*github.com/stretchr/testify/require.Assertions).Contains",
+	//"(*github.com/stretchr/testify/require.Assertions).Equal",
+	//"(*github.com/stretchr/testify/require.Assertions).NotEqual",
+	//"github.com/stretchr/testify/assert.Contains",
+	//"github.com/stretchr/testify/assert.Equal",
+	//"github.com/stretchr/testify/assert.NotEqual",
+	//"github.com/stretchr/testify/require.Contains",
+	//"github.com/stretchr/testify/require.Equal",
+	//"github.com/stretchr/testify/require.NotEqual",
+	)
+)
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	callerPkg := pass.Pkg.Path()
+	for _, allowedPkg := range allowedCallerPackages {
+		if strings.HasPrefix(callerPkg, allowedPkg) {
+			return nil, nil
+		}
+	}
+	inspectResult := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	nodeFilter := []ast.Node{
+		(*ast.CallExpr)(nil),
+	}
+	inspectResult.Preorder(nodeFilter, func(n ast.Node) {
+		call := n.(*ast.CallExpr)
+		fn, ok := typeutil.Callee(pass.TypesInfo, call).(*types.Func)
+		if !ok {
+			return
+		}
+		name := fn.FullName()
+		isAssert := bannedAssertFunctions.Contains(name)
```

Comment:
How about having two variables: `isBannedAssert` and `isBannedEqual` as condition elements for the "skip call" block below?
It would bring symmetry in the code structure.